### ISSUE TITLE
Framework not found App

### ISSUE THREAD

**Issue by XiaoAnXA** [flutter/flutter#62177]
hello，When I was debugging, my mac application was running normally, but the following error occurred when I packaged it, and then I created a new application, and the packaging was normal. I think there is no problem in my environment, but my project appeared I can't solve this error, please help me, thanks

**Expected results:** 

192:im_chat_desktop dliyun$ flutter build macos
** BUILD FAILED **                                                      
                                                                                            
The following build commands failed:                                    
        CodeSign /Users/dliyun/workspace/im_chat_desktop/build/macos/Build/Products/Release/im_char_desktop.app/Contents/Frameworks/App.framework/Versions/A
(1 failure)                                                             
Building macOS application...                                           
Build process failed

```
[        ]     /usr/bin/codesign --force --sign 5EF25EDF039BD7D2FF551F57E457BB9942A8FD49 --timestamp=none --preserve-metadata=identifier,entitlements,flags
/Users/dliyun/workspace/im_chat_desktop/build/macos/Build/Products/Release/im_char_desktop.app/Contents/Frameworks/App.framework/Versions/A
[   +6 ms] /Users/dliyun/workspace/im_chat_desktop/build/macos/Build/Products/Release/im_char_desktop.app/Contents/Frameworks/App.framework/Versions/A: bundle format
unrecognized, invalid, or unsuitable
[        ] Command CodeSign failed with a nonzero exit code
[   +1 ms] ** BUILD FAILED **
[   +5 ms] The following build commands failed:
[        ]      CodeSign /Users/dliyun/workspace/im_chat_desktop/build/macos/Build/Products/Release/im_char_desktop.app/Contents/Frameworks/App.framework/Versions/A
[   +1 ms] (1 failure)
[  +58 ms] Building macOS application... (completed in 13.1s)
[  +10 ms] "flutter macos" took 15,629ms.
[  +11 ms] Build process failed
[        ] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
           #1      buildMacOS (package:flutter_tools/src/macos/build_macos.dart:98:5)
           #2      _rootRunUnary (dart:async/zone.dart:1198:47)
           #3      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #4      _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #5      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #6      Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #7      Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #8      _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
           #9      _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
           #10     _DefaultProcessUtils.stream (package:flutter_tools/src/base/process.dart)
           #11     _rootRunUnary (dart:async/zone.dart:1198:47)
           #12     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #13     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #14     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #15     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #16     Future._addListener.<anonymous closure> (dart:async/future_impl.dart:393:9)
           #17     _rootRun (dart:async/zone.dart:1190:13)
           #18     _CustomZone.run (dart:async/zone.dart:1093:19)
           #19     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
           #20     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
           #21     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
           #22     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
           #23     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
           #24     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)
           
           
[ +294 ms] ensureAnalyticsSent: 264ms
[   +3 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1

```

192:im_chat_desktop dliyun$ flutter doctor -v
[✓] Flutter (Channel master, 1.21.0-5.0.pre, on Mac OS X 10.15.4 19E287, locale zh-Hans-CN)
    • Flutter version 1.21.0-5.0.pre at /Users/dliyun/software/Flutter
    • Framework revision 54c9441723 (6 days ago), 2020-07-17 18:56:12 -0400
    • Engine revision d327828632
    • Dart version 2.9.0 (build 2.9.0-21.0.dev 9dca49e71e)
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/dliyun/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/dliyun/Library/Android/sdk
    • Java binary at: /Library/Java/JavaVirtualMachines/jdk1.8.0_251.jdk/Contents/Home/bin/java
    • Java version Java(TM) SE Runtime Environment (build 1.8.0_251-b08)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.1

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[!] IntelliJ IDEA Ultimate Edition (version 2020.1.1)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • For information about installing plugins, see
      https://flutter.dev/intellij-setup/#installing-the-plugins

[✓] IntelliJ IDEA Community Edition (version 2020.1.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 47.1.3
    • Dart plugin version 201.7223.99

[✓] VS Code (version 1.47.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.12.2

[✓] Connected device (1 available)
    • macOS (desktop) • macos • darwin-x64 • Mac OS X 10.15.4 19E287

! Doctor found issues in 2 categories.

**Comment by pulyaevskiy** [comment]
@XiaoAnXA how did you resolve this issue?

**Comment by ditistijn** [comment]
@XiaoAnXA same problem here. what did you do to resolve this?

**Comment by pulyaevskiy** [comment]
Ok, so in my case this was caused by a weird Pub cache issue. I imported a package in my project which was not declared as a dependency in `pubspec.yaml`. Strangely enough this worked in debug builds, which is why I got confused.

Said package was probably imported as a transitive dependency at some point...

Solution in my case: delete `.dart_tool` directory and run `pub get`, this highlight the missing package. After that release builds work again.

**Comment by ditistijn** [comment]
oh well, maybe I shouldn't have ran my production xcworkspace through xcode 12 beta.
here's my flutter doctor -v:

```
[✓] Flutter (Channel master, 1.21.0-10.0.pre.114, on Mac OS X 10.15.6 19G73, locale nl-NL)
    • Flutter version 1.21.0-10.0.pre.114 at /Users/tijncoertjens/Developer/flutter
    • Framework revision b2c0970aa7 (10 hours ago), 2020-08-16 06:11:02 -0400
    • Engine revision b300be3df3
    • Dart version 2.10.0 (build 2.10.0-34.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at /Users/tijncoertjens/Library/Android/sdk
    • Platform android-30, build-tools 30.0.1
    • Java binary at: /Users/tijncoertjens/Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Users/tijncoertjens/Applications/Android Studio.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.48.0)
    • VS Code at /Users/tijncoertjens/Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.13.2

[✓] Connected device (4 available)
    • iPhone 11 Pro (mobile) • 40176644-EAEB-47E9-9481-35D65403CA37 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-13-6 (simulator)
    • macOS (desktop)        • macos                                • darwin-x64     • Mac OS X 10.15.6 19G73
    • Web Server (web)       • web-server                           • web-javascript • Flutter Tools
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 84.0.4147.125

• No issues found!
```
and the failure:

```
[  +34 ms] ** BUILD FAILED **
[   +1 ms] The following build commands failed:
[        ]      Ld /Users/tijncoertjens/Developer/FlutterProjects/ervaren_in_bewegen/build/macos/Build/Products/Debug/ervaren_in_bewegen.app/Contents/MacOS/ervaren_in_bewegen normal x86_64
[        ] (1 failure)
[  +36 ms] Building macOS application... (completed in 178,4s)
[   +2 ms] Exception: Build process failed
[   +1 ms] "flutter run" took 201.201ms.
[   +5 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:606:7)
           <asynchronous suspension>
           #2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:907:18)
           #3      _rootRunUnary (dart:async/zone.dart:1198:47)
           #4      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #5      _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #6      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #7      Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #8      Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #9      _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
           #10     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:311:13)
           #11     RunCommand.usageValues (package:flutter_tools/src/commands/run.dart)
           #12     _rootRunUnary (dart:async/zone.dart:1198:47)
           #13     _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #14     _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #15     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #16     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #17     Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #18     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:567:7)
           #19     _rootRun (dart:async/zone.dart:1190:13)
           #20     _CustomZone.run (dart:async/zone.dart:1093:19)
           #21     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
           #22     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
           #23     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
           #24     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
           #25     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
           #26     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)
           
           
[ +121 ms] ensureAnalyticsSent: 119ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hook priority 4
[   +1 ms] Shutdown hooks complete
[        ] exiting with code 1
```

**Comment by pulyaevskiy** [comment]
One suggestion that I can add, when you see this error of codesign failing because of "bundle format" - run it again with `--verbose` flag and check the output for any other stack traces before the codesign error.

Just had another issue causing this error and I was able to find an exception stack trace on the step of tree-shaking of one of my fonts. It is weird why it didn't exit at that earlier step but the error was there and I was able to fix it.
